File: Snippets/Fibonacci/fib.pl
#!/opt/perl/bin/perl

use 5.032;

use strict;
use warnings;
no  warnings 'syntax';

use experimental 'signatures';
use experimental 'lexical_subs';


sub is_fib ($n) {
    state $fib = {0 => 1, 1 => 1};
    state $f   = 0;
    state $g   = 1;
    while ($g < $n) {
        ($f, $g)   = ($g, $f + $g);
        $$fib {$g} = 1;
    }
    $$fib {$n}
}


while (<>) {
    say is_fib (0 + $_) || 0;
}

File: Snippets/Fibonacci/fib.py
fib            = {}
fib_prev       = 0
fib_last       = 1
fib [fib_prev] = True;
fib [fib_last] = True;

def is_fib (n):
    global fib, fib_prev, fib_last
    while fib_last < n:
        t        = fib_last
        fib_last = fib_last + fib_prev
        fib_prev = t
        fib [fib_last] = True;
    return n in fib
